tests/phpunit/integration/Core/Authentication/Clients/OAuth_ClientTest.php

```diff
@@ -83,6 +96,10 @@ function ( Request $request ) {
 		$client->refresh_token();
 
 		$this->assertEmpty( get_user_option( OAuth_Client::OPTION_ERROR_CODE, $user_id ) );
+
+		// Verify that the active consumers meta was deleted.
+		$this->assertEmpty( $active_consumers->get() );
```

Comment:
Although this is a good assertion to make here, there are in fact two calls to `$client->refresh_token()` in this test, so it's not currently clear whether the active consumers were cleared as a result of the first or the second call. Of course it's the second call which does clear them, so it would be advisable to add a check after the _first_ call to verify that the active consumers had _not_ been cleared at this point.

In other words, after the first call to `$client->refresh_token()` (currently on line 57), add something like this:

```php
		// Verify the active consumers meta was not deleted.
		$this->assertEquals(
			array(
				1 => array( 'editor', 'author' ),
				2 => array( 'contributor', 'editor' ),
			),
			$active_consumers->get()
		);
```

Reply:
Additionally, it feels slightly inconsistent to be testing for active consumers being cleared in this test, while not actually testing that active consumers were sent with the request.

One way to address this could be adding a check to the fake HTTP client request handler to ensure `active_consumers` is correctly set on the body. In other words something like this:

```php
		$activity_metrics = new Activity_Metrics( $context, $user_options );

		// ...
		$fake_http_client->set_request_handler(
			function ( Request $request ) use ( $activity_metrics ) {
				if ( 0 !== strpos( $request->getUrl(), 'https://oauth2.googleapis.com/token' ) ) {
					return new Response( 200 );
				}

				$body = Query::parse( $request->getBody() );

				// Ensure the token refresh request contains the set of active consumers.
				if ( $activity_metrics->get_for_refresh_token()['active_consumers'] !== $body['active_consumers'] ) {
					return new Response( 200 );
				}

				// ...
```

Reply:
Very valid suggestions. I've addressed them. Thank you, @techanvil!

Reply:
Thanks @nfmohit!